src/Components/Admin/ObjectEditor/object.js

```diff
@@ -0,0 +1,318 @@
+import React, { useEffect, useState } from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import { useErrorHandler } from "react-error-boundary";
+import Select from "react-select";
+import AddIcon from "@material-ui/icons/Add";
+import { Button, IconButton } from "@material-ui/core";
+import {
+    DeleteForever,
+    KeyboardArrowDown,
+    KeyboardArrowUp,
+} from "@material-ui/icons";
+
+import { getPuzzle, editPuzzle } from "../../../lib/puzzleEndpoints";
+
+const useStyles = makeStyles((theme) => ({
+    page: {
+        marginTop: theme.spacing(8),
+    },
+    sceneAndTransitionContainer: {
+        minWidth: 600,
+    },
+    introContainer: {
+        minWidth: 850,
+        display: "flex",
+    },
+    container: {
+        paddingTop: theme.spacing(12),
+    },
+    dragAndDropContainer: {
+        display: "flex",
+        alignItems: "center",
+        marginLeft: "16px",
+    },
+    emptyButtonsContainer: {
+        display: "flex",
+        width: "100%",
+        flexDirection: "column",
+        flexGrow: 1,
+        justifyContent: "center",
+        alignItems: "space-around",
+    },
+    buttonContainer: {
+        display: "flex",
+        width: "100%",
+        justifyContent: "center",
+        marginTop: theme.spacing(12),
+        marginBottom: theme.spacing(4),
+    },
+    button: {
+        display: "flex",
+        justifyContent: "center",
+        padding: "16px 12px",
+        backgroundColor: "#E2E5ED",
+        borderRadius: "12px",
+    },
+}));
+
+class TextPaneView extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            texts: this.props.texts,
+        };
+        const tempTexts = this.props.texts;
+        for (let i = 0; i < tempTexts.length; i++) {
+            tempTexts[i]["index"] = i;
+        }
```

Comment:
Let's try to avoid the for(let...) loops in JS if possible. Can we convert this into a JS function using `map`?